rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -158,15 +158,33 @@ unsafe fn mark_stack<M: Memory>(mem: &mut M, heap_base: u32) {
 }
 
 unsafe fn mark_fields<M: Memory>(mem: &mut M, obj: *mut Obj, obj_tag: Tag, heap_base: u32) {
-    visit_pointer_fields(obj, obj_tag, heap_base as usize, |field_addr| {
-        let field_value = *field_addr;
-        mark_object(mem, field_value);
-
-        // Thread if backwards or self pointer
-        if field_value.get_ptr() <= obj as usize {
-            thread(field_addr);
-        }
-    });
+    visit_pointer_fields(
+        mem,
+        obj,
+        obj_tag,
+        heap_base as usize,
+        |mem, field_addr| {
+            let field_value = *field_addr;
+            mark_object(mem, field_value);
+
+            // Thread if backwards or self pointer
+            if field_value.get_ptr() <= obj as usize {
+                thread(field_addr);
+            }
+        },
+        |mem, slice_start, arr| {
+            const SLICE_INCREMENT: u32 = 127;
+            debug_assert!(SLICE_INCREMENT >= TAG_ARRAY_SLICE_LOW_LIMIT);
```

Comment:
Could that be a static assert at top of file?

Reply:
I could be, but we currently don't have static asserts in the RTS. I don't think it is super urgent.